### PR TITLE
various: update livecheck

### PR DESCRIPTION
**Pull request by khipp**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [ ] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [ ] `brew audit --cask --new <cask>` worked successfully.
- [ ] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [ ] `brew uninstall --cask <cask>` worked successfully.

---

Related to #189875.

### CODE REVIEW

**Review by samford — COMMENTED** [review]
Thanks for catching these, @khipp!

The existing RuboCop unfortunately doesn't check regexes in `strategy` blocks (yet!). It also appears to miss regexes that use interpolation, since the AST is different than non-interpolated regexes.

Outside of the architectural changes to make the `livecheck` block cops apply to casks, part of the challenge is accounting for common cask scenarios that happen infrequently (or not at all) in formulae. The area I'm working on right now is handling interpolated objects (e.g., `url` strings with an interpolated `version`), so this is timely. I'll update the regex cop to handle interpolated regexes while I'm at it 👍

File: Casks/t/toshiba-color-mfp.rb [hunk 1]
```diff
@@ -9,8 +9,9 @@
...
   livecheck do
     url "https://business.toshiba.com/support/downloads/GetDownloads.jsp?model=e-STUDIO3515AC"
-    strategy :page_match do |page|
-      match = page.match(/"MacDC",.*?"id":"(\d+)",.*?"versionName":"(\d+(?:\.\d+)+)",/)
+    regex(/"MacDC",.*?"id":"(\d+)",.*?"versionName":"(\d+(?:\.\d+)+)",/i)
+    strategy :page_match do |page, regex|
+      match = page.match(regex)
       next if match.blank?
 
       "#{match[2]},#{match[1]}"
```
[the remark this answers is not in the document]
Comment (samford) [reply Casks/t/toshiba-color-mfp.rb]: I forgot to remove the `regex` argument from the `strategy` block in that suggestion (sorry about that) but it looks like you caught that right before I was going to mention it 😄

**Comment by khipp** [comment]
@samford Thank you, as always, for the reviews. I’m looking forward to the new audits for `livecheck` blocks!

**Review by samford — APPROVED** [review]
Not sure why but the `toshiba-color-mfp` `Json` check isn't giving me a `ParserError` without the `#strip` change now. So strange. I'm going to convert the brew PR to a draft and test it periodically to see if it pops up again.

Anyway, thanks!

**Pull request merged**